hw/drivers/chg_ctrl/da1469x_charger/include/da1469x_charger/da1469x_charger.h

```diff
@@ -0,0 +1,269 @@
+    uint32_t tbat_comp_timer_valid:1;
+    uint32_t thot_comp_timer_valid:1;
+    uint32_t pwr_up_timer_valid:1;
+
+    uint32_t ctrl;
+    uint32_t voltage_param;
+    uint32_t current_param;
+    uint32_t tempset_param;
+    uint16_t pre_charge_timer;
+    uint16_t cc_charge_timer;
+    uint16_t cv_charge_timer;
+    uint16_t total_charge_timer;
+    uint16_t jeita_v_charge;
+    uint16_t jeita_v_precharge;
+    uint16_t jeita_v_replenish;
+    uint16_t jeita_v_ovp;
+    uint32_t jeita_current;
+    uint16_t vbat_comp_timer;
+    uint16_t vovp_comp_timer;
+    uint16_t tdie_comp_timer;
+    uint16_t tbat_mon_timer;
+    uint16_t tbat_comp_timer;
+    uint16_t thot_comp_timer;
+    uint16_t pwr_up_timer;
+};
+
+struct da1469x_charger_dev {
+    struct os_dev dev;
+#if MYNEWT_VAL(DA1469X_CHARGER_USE_CHARGE_CONTROL)
+    struct charge_control   chg_ctrl;
+#endif
+};
+
+#define DA1469X_ENCODE_V(v) \
+    (((v) < 3800) ? (((v) - 2800) / 50) : \
+                    (((v) < 4600) ? ((v) - 3800) / 20 + 20 : \
+                                    ((v) - 4600) / 100 + 60))
+
+#define DA1469X_ENCODE_CHG_I(i) \
+    (uint16_t)(((i) < 85) ? ((i) / 5) - 1 : \
+                  ((i) < 250 ? ((i) / 10) - 8 + 15 : \
+                               ((i) / 20) - 12 + 31))
+
+#define DA1469X_ENCODE_PRECHG_I(i) \
+    ((uint16_t)((i) < 9 ? (i) * 2 - 1 : \
+                (i) < 25 ? (i) - 8 + 15 : \
+                           (i) / 2 - 12 + 31) << \
+        CHARGER_CHARGER_CURRENT_PARAM_REG_I_PRECHARGE_Pos)
+
+#define DA1469X_ENCODE_EOC_I(i) \
+    ((uint16_t)((i) <= 10 ? ((i) - 4) * 2 / 3 : \
+                    (i) < 18 ? 4 + ((i) - 10) / 2 : \
+                    ((((i) - 12) / 4) + 8)) << \
+        CHARGER_CHARGER_CURRENT_PARAM_REG_I_END_OF_CHARGE_Pos)
+
+typedef enum {
+    CHARGER_STATE_POWER_UP,
+    CHARGER_STATE_INIT,
+    CHARGER_STATE_DISABLED,
+    CHARGER_STATE_PRE_CHARGE,
+    CHARGER_STATE_CC_CHARGE,
+    CHARGER_STATE_CV_CHARGE,
+    CHARGER_STATE_END_OF_CHARGE,
+    CHARGER_STATE_TDIE_PROT,
+    CHARGER_STATE_TBAT_PROT,
+    CHARGER_STATE_BYPASS,
+    CHARGER_STATE_ERROR,
+} charger_state_t;
+
+typedef enum {
+    CHARGER_STATE_IRQ_NONE                      = 0x0000U,
+    CHARGER_STATE_IRQ_DISABLED_TO_PRECHARGE     = 0x0001U,
+    CHARGER_STATE_IRQ_PRECHARGE_TO_CC           = 0x0002U,
+    CHARGER_STATE_IRQ_CC_TO_CV                  = 0x0004U,
+    CHARGER_STATE_IRQ_CC_TO_EOC                 = 0x0008U,
+    CHARGER_STATE_IRQ_CV_TO_EOC                 = 0x0010U,
+    CHARGER_STATE_IRQ_EOC_TO_PRECHARGE          = 0x0020U,
+    CHARGER_STATE_IRQ_TDIE_PROT_TO_PRECHARGE    = 0x0040U,
+    CHARGER_STATE_IRQ_TBAT_PROT_TO_PRECHARGE    = 0x0080U,
+    CHARGER_STATE_IRQ_TBAT_STATUS_UPDATE        = 0x0100U,
+    CHARGER_STATE_IRQ_CV_TO_CC_IRQ              = 0x0200U,
+    CHARGER_STATE_IRQ_CC_TO_PRECHARGE           = 0x0400U,
+    CHARGER_STATE_IRQ_CV_TO_PRECHARGE           = 0x0800U,
+    CHARGER_STATE_IRQ_ALL                       = 0x0FFFU,
```

Comment:
Enums should reflect the fact that these are for da1469x_charger driver, and are not for generic.